## Задача

Подкинули монету N раз. Кол-во случаев, когда выпал орёл, на 10% больше, чем кол-во случаев, когда выпала решка. При каком N мы можем сказать, что монета «нечестная» (орёл и решка выпадают с разной вероятностью)?


## Решение

Допустим, говорить о том, что монетка «нечестная», мы будем на основе Z-критерия для доли (можно использовать нормальную аппроксимацию для биномиального распределения в силу ЦПТ) на уровне значимости $\alpha = 0.05$. Тогда:

$X^n = (X_1, ..., X_n), X \sim Ber(p)$

$H_0: p = p_0$

$H_1: p \neq p_0$

$Z = \frac{\overline{p} - p_0}{\sqrt{\frac{p_0(1-p_0)}{n}}}$

Если $Z > Z_{critical}$, то отвергаем нулевую гипотезу о «честности» монетки в пользу двусторонней альтернативы на уровне значимости $\alpha$.

Тогда минимальное $n$ достигается при $Z = Z_{critical}$ :

$n = \frac{p_0 \cdot (1-p_0)}{(\overline{p}-p_0)^2} \cdot Z_{critical}^2$


In [1]:
import math
import scipy.stats as stats

In [2]:
alpha = 0.05
p0 = 0.5
p = 1/2.1

In [3]:
assert p + 1.1*p == 1

In [4]:
z_critical = stats.norm.ppf(1 - alpha/2)

In [5]:
n_minimal = math.ceil(z_critical**2 * p0*(1-p0) / (p-p0)**2)
print(n_minimal)

1695


**Ответ**: Для определения «честности» монетки с помощью вышеуказанных критерия и уровня значимости нам нужно подкинуть ее как минимум 1695 раз.